doc/src/performance_tuning.rst

```diff
@@ -0,0 +1,151 @@
+.. highlight:: none
+
+******************
+Performance Tuning
+******************
+
+OpenMP Backend
+==============
+
+libxsmm
+-------
+
+Although PyFR will usually function without libxsmm it is nevertheless
+*highly* recommended and can have a very large impact on performance.
+
+AVX-512
+-------
+
+When running on an AVX-512 capable CPU Clang and GCC will, by default,
+only make use of 256-bit vectors.  Given that the kernels in PyFR
+benefit meaningfully from longer vectors it is desirable to override
+this behaviour.  This can be accomplished through the ``cflags`` key
+as::
+
+        [backend-openmp]
+        cflags = -mprefer-vector-width=512
+
+Cores vs. threads
+-----------------
+
+PyFR does not typically derive any benefit from SMT.  As such the
+number of OpenMP threads should be chosen to be equal to the number of
+physical cores.
+
+MPI processes vs. OpenMP threads
+--------------------------------
+
+When using the OpenMP backend it is recommended to employ *one MPI rank
+per NUMA zone*.  For most systems each socket represents its own NUMA
+zone.  Thus, on a two socket system it is suggested to run PyFR with
+two MPI ranks, with each process being bound to a single socket.  The
+specifics of how to accomplish this depend on both the job scheduler
+and MPI distribution.
+
+CUDA Backend
+============
+
+CUDA-aware MPI
+--------------
+
+PyFR is capable of taking advantage of CUDA-aware MPI.  This enables
+CUDA device pointers to be directly passed MPI routines.  Under the
+right circumstances this can result in improved performance for
+simulations which are near the strong scaling limit.  Assuming
+mpi4py has been built against an MPI distribution which is CUDA-aware
+this functionality can be enabled through the ``mpi-type`` key as::
+
+        [backend-cuda]
+        mpi-type = cuda-aware
+
+Note that if `UCX <https://www.openucx.org/>`_ is used as a transport,
+as is the case for recent builds of OpenMPI, it may be necessary to
+set::
+
+        $ export UCX_MEMTYPE_CACHE=n
+
+Partitioning
+============
+
+METIS vs SCOTCH
+---------------
+
+The partitioning module in PyFR includes support for both METIS and
+SCOTCH.  Both usually result in high-quality decompositions.  However,
+for long running simulations on complex geometries it may be worth
+partitioning a grid with both and observing which decomposition
+performs best.
+
+Heterogeneous grids
+-------------------
+
+When running PyFR in parallel on heterogeneous grids it is necessary
```

Comment:
I'm not sure the term heterogenous grids, in this context, is widely used or understood. My first thought for example was to think of heterogeneity in terms of the element size. Maybe add a description of how you are using this term here.

Reply:
Updated to mixed grids